services/github/github-tag.spec.js

```diff
@@ -53,4 +53,24 @@ describe('GithubTag', function () {
       color: 'blue',
     })
   })
+
+  test(GithubTag.getLimit, () => {
+    given({ sort: 'date', filter: undefined }).expect(1)
+    given({ sort: 'date', filter: '' }).expect(1)
+    given({ sort: 'date', filter: '!*-dev' }).expect(100)
+    given({ sort: 'semver', filter: undefined }).expect(100)
+    given({ sort: 'semver', filter: '' }).expect(100)
+    given({ sort: 'semver', filter: '!*-dev' }).expect(100)
+  })
+
+  test(GithubTag.applyFilter, () => {
+    const tags = ['v1.1.0', 'v1.2.0', 'server-2022-01-01']
+    given({ tags, filter: undefined }).expect(tags)
+    given({ tags, filter: '' }).expect(tags)
+    given({ tags, filter: '*' }).expect(tags)
+    given({ tags, filter: '!*' }).expect([])
+    given({ tags, filter: 'foo' }).expect([])
+    given({ tags, filter: 'server-*' }).expect(['server-2022-01-01'])
+    given({ tags, filter: '!server-*' }).expect(['v1.1.0', 'v1.2.0'])
```

Comment:
I haven't really gone to town on testing different filters here. `matcher` has its own test suite.

I also haven't added any additional tests at the service layer. Just unit tests.

Reply:
I agree with the approach, we don't need to re-test the internals of a 3rd party library, just a couple of integration-like tests that give us confidence we're using that lib correctly.

Think you've fully cleared that threshold here :+1: